components/opensea/sources/events/events.mjs

```diff
@@ -0,0 +1,55 @@
+import { axios } from "@pipedream/platform";
+import opensea from "../../opensea.app.mjs";
+
+export default {
+  name: "New OpenSea Collection Events",
+  version: "0.0.1",
+  key: "opensea-events",
+  description: "Emit new filtered OpenSea events",
+  props: {
+    opensea,
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60,
+      },
+    },
+    contractAddress: {
+      type: "string",
+      label: "Contract Address",
+      description: "Collection contract address",
+    },
+    eventType: {
+      type: "string",
+      options: [
+        "sales",
+        "listings",
+      ],
+      label: "Event Type",
+      description: "OpenSea event type",
+    },
+  },
+  dedupe: "greatest",
+  type: "source",
+  methods: {},
+  async run({ $ }) {
+    const contract = this.contractAddress;
+    const eventType = this.eventType === "sales"
+      ? "successful"
+      : "created";
+    const apiKey = this.opensea.$auth.api_key;
+    const url = `https://api.opensea.io/api/v1/events?only_opensea=false&asset_contract_address=${contract}&event_type=${eventType}`;
+    const resp = await axios($, {
```

Comment:
I think this API request could support `pagination` using the `cursor` parameter to be able to retrieve all new events, something like:

```javascript
let cursor = null;
do {
  const resp = this.opensea.retrieveEvents({
    ...,
    cursor,
  });
  cursor = resp.cursor;
} while (cursor);
```

Reply:
The api does support pagination, however, a simple loop will retrieve all history events, we need to set some stop condition.

Reply:
OK, figured this out.